libraries/common/effekt.effekt

```diff
@@ -352,14 +357,20 @@ def abs(n: Double): Double =
 extern pure def log(x: Double): Double =
   js "Math.log(${x})"
   chez "(log ${x})"
+  llvm "%z = call %Double @llvm.log.f64(double ${x}) ret %Double %z"
 
 extern pure def log1p(x: Double): Double =
   js "Math.log1p(${x})"
   chez "(log (+ ${x} 1))"
+  llvm """
+    %i = fadd %Double 1.0, ${x}
+    %z = call %Double @llvm.log.f64(double %i) ret %Double %z
+  """
```

Comment:
FWIW, as far as I remember, `log1p` is in libc as a separate function for numerical precision reasons. I know [MLIR has it as a separate intrinsic](https://mlir.llvm.org/docs/Dialects/MathOps/#mathlog1p-mathlog1pop), but I don't know about LLVM.
I'd either use either that ^ or just with a `default { log(1.0 + x) }` instead of the Chez and LLVM externs?

Reply:
`log1p` does not appear in the LLVM docs for intrinsics, so I defaulted to this